### ISSUE TITLE
[UGUI] Line-Height with `rem` is not the same as in web browsers

### ISSUE THREAD

**Issue by Muchaszewski** [ReactUnity/core#71]
The rem scaling is not as in the web browsers. The rem does not work on the top of the element

I will discuss the following text with the only variable as line-height.
```
<span style="line-height: medium;">This is multiline<br/>text that we comapre</span>
```
line-height: 2rem should be computed as 32px

**NOTE That when `<span>` is used with `<br/>` it's not working as intended -- see issue** https://github.com/ReactUnity/core/issues/72 
**NOTE Manualy fixed to size 16px in ReactUnity for fair comparison for all elements** https://github.com/ReactUnity/core/issues/70
![image](https://user-images.githubusercontent.com/5092066/203513881-39054c90-1f23-4e14-b2ce-cb9b4afa55fd.png)

Fixed font size to 16px and content provided via ``{`text`}`` syntax
![image](https://user-images.githubusercontent.com/5092066/203523391-9e32156e-74c8-49a9-9d18-7a9cda495eb9.png)

**4rem**
![image](https://user-images.githubusercontent.com/5092066/203523786-e5177055-ca28-45c3-826a-838f1db530ae.png)

As you can see the top padding of line spacing is not preserved

**Comment by KurtGokhan** [comment]
ReactUnity uses [TextMeshPro's line-height](https://docs.unity3d.com/Packages/com.unity.textmeshpro@3.2/manual/RichTextLineHeight.html) to implement css `line-height`. This does not work exactly like browsers. TextMeshPro's implementation is actually more like `line-spacing`. I considered adding a hack for this, and it may be possible to implement something closer to browsers. Need to investigate though.

**Comment by KurtGokhan** [comment]
Implemented this. Also had to implement font units like `lh` for this. These units read the font metrics from the current font and calculate size based on that. I am not confident that it will work in all cases. But overall, I am happy about the results.

![image](https://user-images.githubusercontent.com/6034931/208301488-3badc4ad-6812-45a4-9c3b-d869596eb157.png)

There are some unsolvable things like there is a space on the bottom when line-height is smaller than font-size.

**Comment by KurtGokhan** [comment]
Implemented it as in the comment above. Released in v0.14.0
